QuantLib/ql/pricingengines/vanilla/fdeuropeanengine.hpp

```diff
@@ -81,8 +81,13 @@ namespace QuantLib {
                                            results_.delta,
                                            results_.gamma);
         results_.additionalResults["priceCurve"] = prices_;
+        if (model.prevDt() > 0.0) {
+            SampledCurve thetaCurve = prices_;
+            thetaCurve.values() = 
+                (prices_.values() - model.prevValue()) / model.prevDt();
+            results_.additionalResults["thetaCurve"] = thetaCurve;
+        }
```

Comment:
Just a thought: instead of relying on the model keeping the previous state, the engine could explicitly roll to right before today's date, extract the values, and roll to today. This would also make it possible to choose the `dt` to be used for the theta independently of the time step used for rollback.